### PR TITLE
Trigger a link check on a resource from the admin

### PR DESCRIPTION
**Pull request by abulte**

### CODE REVIEW

File: js/components/dataset/resource/availability.vue [hunk 1]
```diff
@@ -1,28 +1,58 @@
 <template>
-    <span v-if="availability === 'AVAILABLE'" class="badge bg-green">✓</span>
-    <span v-if="availability === 'NOT_AVAILABLE'" class="badge bg-red">×</span>
-    <span v-if="availability === 'UNKNOWN'" class="badge bg-gray">?</span>
+    <div :title="_('Click to trigger a new check')" v-if="!checking" @click.stop="refresh()">
```
Comment (davidbgk) [inline js/components/dataset/resource/availability.vue]: Even if available?

File: js/components/dataset/resource/availability.vue [hunk 2]
```diff
@@ -1,28 +1,58 @@
...
                     return 'UNKNOWN';
             }
+        },
+        refresh() {
+            this.checking = true;
+            API.datasets.check_dataset_resource({dataset: this.dataset.id, rid: this.resource.id}, res => {
+                this.availability = this.computeAvailability(res.obj['check:available']);
+                this.checking = false;
+            }, err => {
+                console.log('Something went wrong with the check', err);
```
Comment (davidbgk) [inline js/components/dataset/resource/availability.vue]: Set the availability to unknown in that case?

File: js/components/dataset/resource/availability.vue (hunk 1 above)
Comment (abulte) [reply js/components/dataset/resource/availability.vue]: It's far fetched, but maybe we want to be able to flag immediately a non available resource.

File: js/components/dataset/resource/availability.vue (hunk 2 above)
Comment (abulte) [reply js/components/dataset/resource/availability.vue]: 👍

File: js/components/dataset/resource/availability.vue (hunk 1 above)
Comment (davidbgk) [reply js/components/dataset/resource/availability.vue]: I mean, I don't get the point of refreshing the availability of an available resource?

Comment (abulte) [reply js/components/dataset/resource/availability.vue]: This feature is meant to refresh the check status of a resource when an admin knows the status wrong. Like I said it's far fetched, but maybe we know a resource is unavailable and we want the status to reflect that immediately.

Plus, not displaying the feature when the resource is available would add some complexity and I don't think it's worth it.

I see your point though ;-)

Comment (davidbgk) [reply js/components/dataset/resource/availability.vue]: Ok, get it 👍

**Comment by abulte** [comment]
TODO: this feature is almost useless as is, since it still relies on cached result from the link checker. This should force the retrieval from the link checker, whatever the age of the cache.

**Comment by davidbgk** [comment]
About that TODO, I wonder if the cache should be done at the udata-croquemort level or at the croquemort level, both have pros and cons :)

**Comment by abulte** [comment]
I think this is obsolete, please reopen if it's not.

**Pull request closed without merging**